qcodes/logger/logger.py

```diff
@@ -293,6 +293,36 @@ def start_all_logging() -> None:
     start_logger()
 
 
+def conditionally_start_all_logging() -> None:
+    """Start logging if qcodesrc.json setup for it and not in tool.
+
+    This function will start logging if the session is not being executed by
+    a tool such as pytest and either the GUID components and the
+    instrumentation key for telemetry are set or the key
+    ``config.logger.start_logging_on_import`` is ``True``.
+    """
+    def start_logging_on_import() -> bool:
+        config = qc.config
+        return (
+            config.GUID_components.location != 0
+            and config.GUID_components.work_station != 0
+            and config.telemetry.instrumentation_key != \
+                "00000000-0000-0000-0000-000000000000"
+            or config.logger.start_logging_on_import
```

Comment:
I would still consider if it makes sense to make `start_logging_on_import` an enum with 3 states. `Always`, `Never` and `if_setup_for_telemetry`

Reply:
Yes that is right, sorry Jens, I forgot about this point. I am on it.